src/test/java/net/cardosi/mojo/MojoTester.java

```diff
@@ -0,0 +1,136 @@
+            DefaultRepositorySystemSession repoSession =
+                  (DefaultRepositorySystemSession) maven.newRepositorySession( request );
+            repoSession.setLocalRepositoryManager(
+                  new SimpleLocalRepositoryManagerFactory().newInstance(repoSession,
+                                                                        new LocalRepository(request.getLocalRepository().getBasedir() )));
+
+            @SuppressWarnings("deprecation")
+            MavenSession session = new MavenSession( getContainer(),
+                                                     repoSession,
+                                                     request, result );
+            return session;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** As {@link #lookupConfiguredMojo(MavenProject, String)} but taking the pom file
+     * and creating the {@link MavenProject}. */
+    protected Mojo lookupConfiguredMojo(File pom, String goal) throws Exception {
+        assertNotNull( pom );
+        assertTrue( pom.exists() );
+
+        MavenSession session = newMavenSession();
+        session.getRequest().setPom(pom);
+        session.getProjectBuildingRequest().setResolveDependencies(true);
+
+        GraphBuilder gbuilder = getContainer().lookup(GraphBuilder.class);
+        gbuilder.build(session);
+
+        List<MavenProject> mavenProjects = session.getProjects();
+
+        ProjectBuilder projectBuilder = lookup(ProjectBuilder.class);
+        Method method = projectBuilder.getClass().getDeclaredMethod("resolveDependencies", MavenProject.class, RepositorySystemSession.class);
+        method.setAccessible(true);
+
+        for (MavenProject mavenProject : mavenProjects) {
+            method.invoke(projectBuilder, mavenProject, session.getRepositorySession());
+        }
+
+        final MojoExecution mojoExecution = newMojoExecution(goal);
+
+        MojoDescriptor mojoDescriptor = mojoExecution.getMojoDescriptor();
+        PluginDescriptor pluginDescriptor = mojoDescriptor.getPluginDescriptor();
+
+        Plugin plugin = session.getCurrentProject().getBuild().getPluginsAsMap().get(pluginDescriptor.getGroupId() + ":" + pluginDescriptor.getArtifactId());
+        pluginDescriptor.setPlugin(plugin);
+
+        AbstractBuildMojo mojo = (AbstractBuildMojo) lookupConfiguredMojo( session, newMojoExecution( goal ) );
+        Map<String, Object> pluginContext = mojo.mavenSession.getPluginContext(pluginDescriptor, session.getCurrentProject());
+        pluginContext.put( "project", session.getCurrentProject() );
+        pluginContext.put( "pluginDescriptor", pluginDescriptor );
+        ( (ContextEnabled) mojo ).setPluginContext(pluginContext);
+
+        return mojo;
+    }
+
+    @Test
+    @Ignore
+    public void testMojoGoal() throws Exception
+    {
+        File testPom = new File(getBasedir(),
+                                "target/test-classes/hello-world-reactor/pom.xml" );
```

Comment:
if this is targetting just that parent pom, why the changes to the transitive-dependencies parent pom?

the next iter will have the actual building code in a plain java jar library (maven agnostic, so you could build some IDE tooling too in theory), and will be used by the mojo - this tooling will be helpful, but we'll be able to actually write plain tests too, and actually edit some temp files and see what happens.